pkg/apis/sessions/session_state.go

```diff
@@ -149,8 +152,32 @@ func (s *SessionState) GetClaim(claim string) []string {
 	case "preferred_username":
 		return []string{s.PreferredUsername}
 	default:
-		return []string{}
+		return s.getAdditionalClaim(claim)
+	}
+}
+
+func (s *SessionState) getAdditionalClaim(claim string) []string {
+	if value, ok := s.AdditionalClaims[claim]; ok {
+		switch v := value.(type) {
+		case string:
+			return []string{v}
+		case []string:
+			return v
+		case []interface{}:
+			result := make([]string, len(v))
+			for i, item := range v {
+				if str, ok := item.(string); ok {
+					result[i] = str
+				} else {
+					result[i] = fmt.Sprintf("%v", item)
+				}
+			}
+			return result
+		default:
+			return []string{fmt.Sprintf("%v", value)}
+		}
 	}
+	return []string{}
 }
```

Comment:
This reminds me of what we already have in the ClaimExtractor

https://github.com/oauth2-proxy/oauth2-proxy/blob/bae509d2ecde533ffa32a6a72159eb948cabd1d3/pkg/providers/util/claim_extractor.go#L170-L190

This to me looks like a duplication of efforts and I imagine we can somehow marry those two 🤔